utils/prometheus-node-exporter-lua/files/usr/lib/lua/prometheus-collectors/wifi_stations.lua

```diff
@@ -3,10 +3,23 @@ local iwinfo = require "iwinfo"
 
 local function scrape()
   local metric_wifi_stations = metric("wifi_stations", "gauge")
+
   local metric_wifi_station_signal = metric("wifi_station_signal_dbm","gauge")
+
+  local metric_station_inactive = metric('wifi_station_inactive_ms', 'gauge')
```

Comment:
Now all metrics are with wifi_station prefix.

Reply:
> Also not sure how usefull is wifi_station_inactive_ms, if there is not tx/rx, it's inactive
> (can you share your use-case)

Well, typical scrape interval for wifi routers is about 5-10s. But stations can sleep from tens of milliseconds to N seconds. So we can figure out the average sleeping time of  a station if it's less than our scrape interval. Better accuracy for better monitoring.
In my case this feature is helpful for DTIM and U-APSD tuning.

Reply:
Do you have a link that explain the behavior of "inactive time" ?
I imagine that it goes to 0 each time we TX or RX, thus because we poll every 10s data is useless (but I might imagine wrong)
Is it a counter over last X seconds ?

Reply:
Also it would be great to use the same name as the original node exporter
https://github.com/prometheus/node_exporter/blob/master/collector/wifi_linux.go
for example wifi_station_inactive_seconds
wifi_station_receive_bits_per_second
...

Reply:
Yes, sure.
According to this
https://git.openwrt.org/?p=project/iwinfo.git;a=blob;f=iwinfo_nl80211.c;h=200be28d9a44aeb1e95528e10c2f0c63abf13046;hb=refs/heads/master#l1876
Inactive time will always be fetched from NL80211.

It's logic is simple. Just a difference between current time and last rx/ack time of the station.
https://elixir.bootlin.com/linux/v4.20.17/source/net/mac80211/sta_info.c#L2135

It is not a metric for polling station activity, but it's a good instrument to debug stations power-saving techniques. Of course we need to decrease scrape interval to 1-2s for that.
So I think it can be helpful.

Reply:
> Also it would be great to use the same name as the original node exporter
> https://github.com/prometheus/node_exporter/blob/master/collector/wifi_linux.go
> for example wifi_station_inactive_seconds
> wifi_station_receive_bits_per_second
> ...

I renamed all metrics according to the original Linux node exporter.

Reply:
Noise metric will be exported per-dev.